js/classes/game_engine.js

```diff
@@ -203,35 +203,74 @@ class GameEngine {
         }, false);
     };
 
+    /**
+     * This is (the only thing, currently, that is) called when Debug mode is active!
+     * This draws a grid around all blocks, and every 5 block cells are labeled.
+     */
     drawGrid() {
+
+        // (1) Draw the grid: 
+
         CTX.strokeStyle = "white";
         CTX.strokeWeight = 1;
-        // Draw the grid.
-        for (let x = DIMENSION.MIN_X; x <= DIMENSION.MAX_X; x += Block.SCALED_SIZE) {
-            for (let y = DIMENSION.MIN_Y; y <= DIMENSION.MAX_Y; y += Block.SCALED_SIZE) {
-                CTX.beginPath();
-                CTX.moveTo(DIMENSION.MIN_X - CAMERA.pos.x, y - CAMERA.pos.y);
-                CTX.lineTo(DIMENSION.MAX_X - CAMERA.pos.x, y - CAMERA.pos.y);
-                CTX.stroke();
-
-                CTX.beginPath();
-                CTX.moveTo(x - CAMERA.pos.x, DIMENSION.MIN_Y - CAMERA.pos.y);
-                CTX.lineTo(x - CAMERA.pos.x, DIMENSION.MAX_Y - CAMERA.pos.y);
-                CTX.stroke();
-            }
+
+        // Draw all the vertical lines by iterating through the x values we need.
+        for (let x = ZONE.MIN_PT.x; x <= ZONE.MAX_PT.x; x += Block.SCALED_SIZE) {
+            // Start a path
+            CTX.beginPath();
+            // Starting point
+            CTX.moveTo(x - CAMERA.pos.x, ZONE.MIN_PT.y - CAMERA.pos.y);
```

Comment:
I'm a little confused on what moveTo does with the x and y position of camera. Could you explain this?